azext_iot/tests/central/test_iot_central_int.py

```diff
@@ -647,3 +730,11 @@ def _get_monitor_events_output(self, device_id, enqueued_time, asserts=None):
             output = ""
 
         return output
+
+    def _connect_gettwin_disconnect_wait_tobeprovisioned(self, device_id, credentials):
+        device_client = helpers.dps_connect_device(device_id, credentials)
+        assert device_client.connected
```

Comment:
Don't assert until after the test passes or you won't clean up things in the IT pipeline on errors